### PR TITLE
Use the new compileResource API when using Daffodil 3.9.0+

### LINKED ISSUE #47: packageDaffodilBin does not create depersonalized diagnostic paths
A new Daffodil API is required for packageDaffodilBin to create saved parsers with depersonalized paths. See DAFFODIL-2919 for one possible way to achieve this.

In the meantime, one alternative is to set `exportJars := true` in build.sbt. This causes packageDaffodilBin to find the main schema in a jar instead of in a file and allows Daffodil to depersonalize the path.

Once DAFFODIL-2919 is resolved, we should update the plugin to make sure to incorporate the change.

### PR DESCRIPTION
**Pull request by stevedlawrence**
The compileResource API is useful to create depersonalized diagnostics and saved-parsers.

This modifies the DaffodilPlugin to use compileResource instead of compileSource when creating a saved parser for newer versions of Daffodil. DaffodilSaver must still support older versions of Daffodil that do not have this new API, so the plugin passes in a number to the saver to signify which version of the API to use.

Closes #47

### CODE REVIEW

File: src/main/scala/org/apache/daffodil/DaffodilPlugin.scala [hunk 1]
```diff
@@ -361,10 +361,26 @@ object DaffodilPlugin extends AutoPlugin {
               .map(_.split("\\s+").toSeq)
               .getOrElse(Seq.empty)
 
+            // the DaffodilSaver needs to know which version of Daffodil we are using to create
+            // a saved, since some versions of Daffodil have different APIs and it must use the
```
Comment (mbeckerle) [inline src/main/scala/org/apache/daffodil/DaffodilPlugin.scala:365]: "a saved processor"

**Review by mbeckerle — APPROVED** [review]
+1

**Comment by stevedlawrence** [comment]
Rebased onto latest main branch to ensure the 3.9.0 tests run.

**Review by olabusayoT — APPROVED** [review]
+1

**Pull request merged**